packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -0,0 +1,450 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BasePlugin, NoopLogger, isValid } from '@opentelemetry/core';
+import { CanonicalCode, Span, SpanKind, SpanOptions, Logger, Status } from '@opentelemetry/types';
+import { NodeTracer } from '@opentelemetry/node-tracer';
+import { ClientRequest, IncomingMessage, request, get, RequestOptions, ServerResponse } from 'http';
+import * as http from 'http';
+import * as semver from 'semver';
+import * as shimmer from 'shimmer';
```

Comment:
I believe @rochdev said that shimmer is quite expensive in term of performance and since the `http` plugin might be the most used, we might want to use a standard monkey patching approach here.

Reply:
The patch itself is slow, but it happens only once for the plugin, so it should be fine. I would still prefer that we have our own faster shimmer internally, but that can be added later.